Doc/library/pathlib.rst

```diff
@@ -1641,6 +1621,45 @@ call fails (for example because the path doesn't exist).
    .. versionchanged:: 3.10
       The *newline* parameter was added.
 
+
+.. _pattern-language:
+
+Pattern language
+----------------
+
+The following wildcards are supported in patterns for
+:meth:`~PurePath.full_match`, :meth:`~Path.glob` and :meth:`~Path.rglob`:
+
+=========================  ===========================================
+Wildcard                   Matches
+=========================  ===========================================
+"``**``" (full segment)    Any number of file or directory segments.
+"``*``" (full segment)     One file or directory segment.
+"``*``" (otherwise)        Any number of non-separator characters.
+"``?``"                    One non-separator character.
+"``[seq]``"                One character in *seq*.
+"``[!seq]``"               One character not in *seq*.
+=========================  ===========================================
+
+The "``**``" wildcard enables recursive globbing. A few examples:
+
+=========================  ===========================================
+Pattern                    Meaning
+=========================  ===========================================
+"``**/*``"                 Any path with at least one segment.
+"``**/*.py``"              Any path with a final segment ending "``.py``".
+"``assets/**``"            Any path starting with "``assets/``".
+"``assets/**/*``"          Any path starting with "``assets/``", excluding "``assets/``" itself.
+=========================  ===========================================
+
+.. note::
+   Globbing with the "``**``" wildcard in large directory trees may consume
+   an inordinate amount of time.
```

Comment:
"Inordinate" (to me) suggests an unreasonable amount of time. It's not unreasonable, it's just proportionate to the size of the entire tree.

Maybe we can say something like:

> Globbing with the "``**``" wildcard unavoidably has to visit every directory in the tree. Large directory trees may take a long time to search. Consider whether :func:`os.scandir` and a manual recursion algorithm may be more appropriate.

Reply:
Thanks - I agree about "inordinate" and will fix.

> Consider whether :func:`os.scandir` and a manual recursion algorithm may be more appropriate.

`glob()` is so well-optimized now that a user would struggle to get better performance out of a manual `scandir()`-based algorithm that operated on pathlib paths :sweat_smile:.

Reply:
I was more thinking about ridiculous directory trees, and if the structure is roughly known then it may make sense to optimise the search that way.

For example, exclude `.git/` from a recursive glob.